core/client_test.go

```diff
@@ -96,6 +98,139 @@ func TestClientImplementsRequestMaker(t *testing.T) {
 // Perhaps consider golden/testdata integration for generating request/response
 // data.
 
+type fakeRequestMaker2Call struct {
+	Ctx     context.Context
+	Request *katapult.Request
+	V       interface{}
+}
+
+type fakeRequestMaker2Responder func(
```

Comment:
I wonder if we could hand some more logic off of this type, to move it out of the DoIt method. Not sure if it would be tidier or worse, but I'm curious.